### DISCUSSION TITLE
Intergrate with InfluxDB

### DISCUSSION THREAD

**Discussion by dzecevic** [tbnobody/OpenDTU#312] [Show and tell]
I have created tutorial how had i integrated OpenDTU with mosquitto, Telegrapf and InfluxDB.

Link is here: https://www.zecevic-ninja.com/?p=35

Greets
Domagoj

**Comment by meystervie** [comment]
Hi great Tutorial and exactly what I am looking for. 
Unfortunatly I am running a Synology ARM (DS218) not too slow but lacking Docker support.
Do you know if Flux and Telegraf both work on ARM64 Synology?

**Reply by dzecevic** [reply]
Hi!

Thanks!
Best that i can say is that there is some support for ARM devices
https://portal.influxdata.com/downloads/
https://github.com/influxdata/telegraf/releases

So if you manage to install and run them then config should be the same.

Greets
Domagoj

**Comment by joba-1** [comment]
The nas has a recent python, right?
If docker is not an option, you could also try my [openDtu2Db](https://github.com/joba-1/openDtu2Db) python script.
It comes with a systemd service file for starting and stopping the script as a background services. It should be reasonably easy to adapt if needed. 
Good for you, if I had seen this nice telegraf solution earlier, I would probably have used that and not written the script - now you have options :)

**Comment by el-dom** [comment]
Thanks for sharing your solution.

I noticed there are a couple of values that cannot be processed by telegraf. E.g. the `name` of the Inverter and its strings or the buildtimestamp  

```
solar/116164603197/name Shack
solar/116164603197/1/name East1
solar/116164603197/2/name East2
solar/116164603197/device/fwbuilddatetime 2020-06-24 19:00:00
```

They end up in errors like this

```
2022-12-21T13:03:14Z E! [inputs.mqtt_consumer] Error in plugin: strconv.ParseFloat: parsing "Shack": invalid syntax
2022-12-21T13:03:14Z E! [inputs.mqtt_consumer] Error in plugin: strconv.ParseFloat: parsing "East1": invalid syntax
2022-12-21T13:03:14Z E! [inputs.mqtt_consumer] Error in plugin: strconv.ParseFloat: parsing "East2": invalid syntax
2022-12-21T13:03:14Z E! [inputs.mqtt_consumer] Error in plugin: strconv.ParseFloat: parsing "19:00:00": invalid syntax
```

Which kind of makes sense as telegraf was configured to expect float values

```
data_format = "value"
data_type = "float"
```

Were you able to preserve that information or is telegraf discarding it for you as well?
